In [22]:
# Questions adapted from xxx
print("1. What is the total number of people on the titanic and how many of them survived and how many did not?")
print("2. How many that survived were female and how many that died were female.")
print("3. How many children were on the titanic, NB: you are a child if age < 17.")
print("4. How many children died that were on the ship.")
print("5. How many people had families with them.")
print("6. What is the ratio of female to male.")
print("7. What contributed to the survival of those who survived.")

1. What is the total number of people on the titanic and how many of them survived and how many did not?
2. How many that survived were female and how many that died were female.
3. How many children were on the titanic, NB: you are a child if age < 17.
4. How many children died that were on the ship.
5. How many people had families with them.
6. What is the ratio of female to male.
7. What contributed to the survival of those who survived.


In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [13]:
# get and open the file, show 10 rows as a sample

data = pd.read_csv('Titanic_train.csv')
data.head(10)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male   NaN      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
5      0            330877   8.4583   NaN        Q  
6      0             17463  51.8625   E46        S  
7      1            349909  21.0750   NaN        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

In [14]:
data.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [15]:
data['PassengerId']

0        1
1        2
2        3
3        4
4        5
      ... 
886    887
887    888
888    889
889    890
890    891
Name: PassengerId, Length: 891, dtype: int64

In [16]:
data['PassengerId'].max()

891

In [17]:
len(data[data['Survived']==1])

342

1. What is the total number of people on the titanic and how many of them survived and how many did not?
2. How many that survived were female and how many that died were female.
3. How many children were on the titanic, NB: you are a child if age < 17.
4. How many children died that were on the ship.
5. How many people had families with them.
6. What is the ratio of female to male.
7. What contributed to the survival of those who survived.

In [18]:
#Question 1: What is the total number of people on the titanic and how many of them survived and how many did not?
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
PassengerId    891 non-null int64
Survived       891 non-null int64
Pclass         891 non-null int64
Name           891 non-null object
Sex            891 non-null object
Age            714 non-null float64
SibSp          891 non-null int64
Parch          891 non-null int64
Ticket         891 non-null object
Fare           891 non-null float64
Cabin          204 non-null object
Embarked       889 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


Answer: 891 people on the Titanic

In [19]:
print(len(data[data["Survived"] == 0]))
print(len(data[data["Survived"] == 1]))

549
342


Answer: 549 people did not survive and 342 people survived.

In [20]:
# Question 2: How many that survived were female and how many that died were female?
print(len(data[(data["Survived"]==0)&(data["Sex"]=="female")]))
print(len(data[(data["Survived"]==1)&(data["Sex"]=="female")]))

81
233


Answer: The number of females who survied was 233, and the number of females who died was 81.

In [23]:
# Question 3: How many children were on the titanic? NB: you are a child if age < 17.
len(data[data["Age"]<17])

100

Answer: There were 100 children on the Titanic


In [24]:
# Question 4: How many children died that were on the ship?
len(data[(data["Survived"]==0) & (data["Age"]<17)])

45

Answer: 45 children died on the ship.

In [25]:
# Question 5: How many people had families with them?
len(data[data["SibSp"]>0])

283

Answer: 283 people had families or at least a spouse with them.

In [26]:
# Question 6: What is the ratio of female to male?
len(data[data["Sex"]=="female"]) / len(data[data["Sex"]=="male"])

0.5441941074523396

In [27]:
data.Sex.value_counts()

male      577
female    314
Name: Sex, dtype: int64

Answer: The ratio of female to male was 0.5441941074523396

In [28]:
# Question 7: What contributed to the survival of those who survived?

In [32]:
b = pd.get_dummies(data['Embarked'], columns=['New_Embarked'])

In [33]:
c = pd.get_dummies(data['Sex'], columns=['New_Sex'])

In [34]:
df = data.merge(b, how='left',left_index=True, right_index=True)

In [35]:
df = data.merge(c, how='left',left_index=True, right_index=True)

In [37]:
del df['Embarked']

In [38]:
del df['Sex']

In [39]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name   Age  SibSp  Parch  \
0                            Braund, Mr. Owen Harris  22.0      1      0   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  38.0      1      0   
2                             Heikkinen, Miss. Laina  26.0      0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  35.0      1      0   
4                           Allen, Mr. William Henry  35.0      0      0   

             Ticket     Fare Cabin  female  male  
0         A/5 21171   7.2500   NaN       0     1  
1          PC 17599  71.2833   C85       1     0  
2  STON/O2. 3101282   7.9250   NaN       1     0  
3            113803  53.1000  C123       1     0  
4            373450   8.0500   NaN       0     1

In [40]:
df.corr()

PassengerId  Survived    Pclass       Age     SibSp     Parch  \
PassengerId     1.000000 -0.005007 -0.035144  0.036847 -0.057527 -0.001652   
Survived       -0.005007  1.000000 -0.338481 -0.077221 -0.035322  0.081629   
Pclass         -0.035144 -0.338481  1.000000 -0.369226  0.083081  0.018443   
Age             0.036847 -0.077221 -0.369226  1.000000 -0.308247 -0.189119   
SibSp          -0.057527 -0.035322  0.083081 -0.308247  1.000000  0.414838   
Parch          -0.001652  0.081629  0.018443 -0.189119  0.414838  1.000000   
Fare            0.012658  0.257307 -0.549500  0.096067  0.159651  0.216225   
female         -0.042939  0.543351 -0.131900 -0.093254  0.114631  0.245489   
male            0.042939 -0.543351  0.131900  0.093254 -0.114631 -0.245489   

                 Fare    female      male  
PassengerId  0.012658 -0.042939  0.042939  
Survived     0.257307  0.543351 -0.543351  
Pclass      -0.549500 -0.131900  0.131900  
Age          0.096067 -0.093254  0.093254  
SibSp        0.159651  0.114631 -0.114631  
Parch        0.216225  0.245489 -0.245489  
Fare         1.000000  0.182333 -0.182333  
female       0.182333  1.000000 -1.000000  
male        -0.182333 -1.000000  1.000000

Answer: From the above correlation table, we can tell that "Pclass", "Fare", and "Sex" are the columns with the highest correlation with "Survived"